### PR TITLE
init translation for Web Locks API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

see https://developer.mozilla.org/en-US/docs/Web/API/Web_Locks_API

### Related issues and pull requests

part of #17542

### CODE REVIEW

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+### 死锁
+
+当进程的每个部分都在等待无法满足的请求而无法再取得进展时，就会发生死锁。此 API 可能会在复杂的用例中发生这种情况，例如，如果无序请求多个锁。如果选项卡 1 持有锁 A 并且选项卡 2 持有锁 B，接着选项卡 1 也尝试获取锁 B，而选项卡 2 也尝试获取锁 A，则两个请求都不会被授予锁。Web 应用程序可以通过多种策略来避免这种情况，例如确保锁定请求不嵌套、始终有序或设定超时。请注意，此类死锁仅影响锁本身和依赖于它们的代码；浏览器、其他选项卡和页面中的其他脚本不受影响。
+
+## 接口
+
+- {{domxref("Lock")}}
+  - : 提供先前请求的锁的名称和模式，该名称和模式在传入 {{domxref("LockManager.request()")}} 的回调方法中接收。
```
Comment (yin1999) [inline files/zh-cn/web/api/web_locks_api/index.md:104]: https://developer.mozilla.org/en-US/docs/Web/API/LockManager/request#general_example

```suggestion
  - : 提供先前请求的锁的名称和模式，该锁会被传入 {{domxref("LockManager.request()")}} 的回调方法接收。
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- {{domxref("LockManager")}}
+  - : 提供请求新的 {{domxref("Lock")}} 对象和查询现有 {{domxref('Lock')}} 对象的方法。要获取 {{domxref("LockManager")}} 的实例，访问 {{domxref("navigator.locks")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/web_locks_api/index.md:106]:
```suggestion
  - : 提供请求新的 {{domxref("Lock")}} 对象和查询现有 {{domxref('Lock')}} 对象的方法。要获取 {{domxref("LockManager")}} 的实例，请访问 {{domxref("navigator.locks")}}。
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+当持有锁时，来自此执行上下文或其他选项卡以及 Worker 进程的相同锁的请求将排队。仅当锁被释放时，第一个排队的请求才会被授予锁。
+
+API 提供了可根据需要使用的可选功能，包括：
+
+- 从异步任务返回值
+- 共享锁和独占锁模式
+- 有条件获取
+- 诊断以查询源中锁的状态
+- 防止死锁的逃生舱
```
Comment (yin1999) [inline files/zh-cn/web/api/web_locks_api/index.md:41]: This term "escape hatch" is from database design, I'd prefer "应急通道"/"应急方法"

https://docs.polardbpg.com/1696830472553/features/v11/performance/sharedserver.html

```suggestion
- 防止死锁的应急通道
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions:

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+{{DefaultAPISidebar("Web Locks API")}}{{securecontext_header}}
+
+**Web Locks API** 允许在一个选项卡或工作线程中运行的脚本异步获取锁，在执行工作时保持锁，最后释放锁。持有锁时，在同一源中执行的其他脚本都无法获取相同的锁，这允许在多个选项卡或工作线程中运行的 Web 应用程序协调工作和资源的使用。
+
+{{AvailableInWorkers}}
+
+## 概念和用法
+
+锁是一个抽象概念，代表一些潜在的共享资源，由 Web 应用程序选择的名称进行标识。例如，如果在多个选项卡中运行的 Web 应用程序想要确保只有一个选项卡在网络和索引数据库之间同步数据，则每个选项卡都可以尝试获取“my_net_db_sync”锁，但只有一个选项卡会成功（[领导者选举模式](https://en.wikipedia.org/wiki/Leader_election)。）
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/web_locks_api/index.md:16]:
```suggestion
锁是一个抽象概念，代表一些潜在的共享资源，由 Web 应用程序选择的名称进行标识。例如，如果在多个标签页中运行的 Web 应用程序想要确保只有一个标签页在网络和索引数据库之间同步数据，则每个标签页都可以尝试获取“my_net_db_sync”锁，但只有一个标签页会成功（[领导者选举模式](https://en.wikipedia.org/wiki/Leader_election)。）
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+navigator.locks.request(
+  "my_resource",
+  // 锁已被获取。
+  (lock) => p, // 现在锁将被保持直到调用 resolve() 方法或 reject() 方法为止。
+);
+```
+
+### 死锁
+
+当进程的每个部分都在等待无法满足的请求而无法再取得进展时，就会发生死锁。此 API 可能会在复杂的用例中发生这种情况，例如，如果无序请求多个锁。如果选项卡 1 持有锁 A 并且选项卡 2 持有锁 B，接着选项卡 1 也尝试获取锁 B，而选项卡 2 也尝试获取锁 A，则两个请求都不会被授予锁。Web 应用程序可以通过多种策略来避免这种情况，例如确保锁定请求不嵌套、始终有序或设定超时。请注意，此类死锁仅影响锁本身和依赖于它们的代码；浏览器、其他选项卡和页面中的其他脚本不受影响。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/web_locks_api/index.md:99]:
```suggestion
当进程的每个部分都在等待无法满足的请求而无法再取得进展时，就会发生死锁。此 API 可能会在复杂的用例中发生这种情况，例如，如果无序请求多个锁。如果标签页 1 持有锁 A 并且标签页 2 持有锁 B，接着标签页 1 也尝试获取锁 B，而标签页 2 也尝试获取锁 A，则两个请求都不会被授予锁。Web 应用程序可以通过多种策略来避免这种情况，例如确保锁定请求不嵌套、始终有序或设定超时。请注意，此类死锁仅影响锁本身和依赖于它们的代码；浏览器、其他标签页和页面中的其他脚本不受影响。
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+```js
+navigator.locks.request("my_resource", async (lock) => {
+  // 锁已被获取。
+  await do_something();
+  await do_something_else();
+  // 现在锁将被释放。
+});
+```
+
+当持有锁时，来自此执行上下文或其他选项卡以及 Worker 进程的相同锁的请求将排队。仅当锁被释放时，第一个排队的请求才会被授予锁。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/web_locks_api/index.md:33]:
```suggestion
当持有锁时，来自此执行上下文或其他标签页以及 Worker 进程的相同锁的请求将排队。仅当锁被释放时，第一个排队的请求才会被授予锁。
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+- `steal`：如果指定，则将释放任何持有的同名锁，并且请求将被授予，抢占任何排队的请求。
+- `signal`：可以传入一个 {{domxref("AbortSignal")}}，允许中止锁定请求。这可用于实现请求超时。
+
+### 监控
+
+脚本可以使用 {{domxref("LockManager.query", "navigator.locks.query()")}} 方法来内省源锁管理器的状态。这在调试时非常有用，例如，确定无法获取锁的原因。结果是一个锁管理器状态的快照，它标识了拍摄快照时持有的锁和请求中的锁以及有关每个锁的一些附加数据（例如模式）。
+
+### 进阶使用
+
+对于更复杂的情况，例如在任意时间内保持锁，回调可以返回一个脚本显式解析的 Promise：
```
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/api/web_locks_api/index.md:79]: see the [documentation](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#:~:text=%E4%BD%A0%E8%BF%98%E4%BC%9A%E5%90%AC%E5%88%B0%E4%BD%BF%E7%94%A8%E5%B7%B2%E8%A7%A3%E5%86%B3%EF%BC%88resolved%EF%BC%89%E8%BF%99%E4%B8%AA%E6%9C%AF%E8%AF%AD%E6%9D%A5%E6%8F%8F%E8%BF%B0)

```suggestion
对于更复杂的情况，例如在任意时间内保持锁，回调可以返回一个脚本显式解决的 Promise：
```

**Review by yin1999 — COMMENTED** [review]
The last one:

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+- `steal`：如果指定，则将释放任何持有的同名锁，并且请求将被授予，抢占任何排队的请求。
+- `signal`：可以传入一个 {{domxref("AbortSignal")}}，允许中止锁定请求。这可用于实现请求超时。
+
+### 监控
+
+脚本可以使用 {{domxref("LockManager.query", "navigator.locks.query()")}} 方法来内省源锁管理器的状态。这在调试时非常有用，例如，确定无法获取锁的原因。结果是一个锁管理器状态的快照，它标识了拍摄快照时持有的锁和请求中的锁以及有关每个锁的一些附加数据（例如模式）。
+
+### 进阶用法
+
+对于更复杂的情况，例如在任意时间内保持锁，回调可以返回一个脚本显式解析的 Promise：
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/web_locks_api/index.md:79]:
```suggestion
对于更复杂的情况，例如在任意时间内保持锁，回调方法可以返回一个由脚本显式解析的 Promise：
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+- `steal`：如果指定，则将释放任何持有的同名锁，并且请求将被授予，抢占任何排队的请求。
+- `signal`：可以传入一个 {{domxref("AbortSignal")}}，允许中止锁定请求。这可用于实现请求超时。
+
+### 监控
+
+脚本可以使用 {{domxref("LockManager.query", "navigator.locks.query()")}} 方法来内省源锁管理器的状态。这在调试时非常有用，例如，确定无法获取锁的原因。结果是一个锁管理器状态的快照，它标识了拍摄快照时持有的锁和请求中的锁以及有关每个锁的一些附加数据（例如模式）。
+
+### 进阶用法
+
+对于更复杂的情况，例如在任意时间内保持锁，回调方法可以返回一个由脚本显式解析的 Promise：
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/web_locks_api/index.md:79]:
```suggestion
对于更复杂的情况，例如在任意时间内保持锁，回调可以返回一个脚本显式解决的 Promise：
```

File: files/zh-cn/web/api/web_locks_api/index.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+API 提供了可根据需要使用的可选功能，包括：
+
+- 从异步任务返回值
+- 共享锁和独占锁模式
+- 条件获取
+- 诊断以查询源中锁的状态
+- 防止死锁的应急通道
+
+锁的范围仅限于同一个源内；一个选项卡从 `https://example.com` 获取的锁对另一个选项卡从 `https://example.org:8080` 获取的锁没有影响，因为它们是不同的源。
```
Comment (yin1999) [inline files/zh-cn/web/api/web_locks_api/index.md:43]:
```suggestion
锁的范围仅限于同一个源内；一个标签页从 `https://example.com` 获取的锁对另一个标签页从 `https://example.org:8080` 获取的锁没有影响，因为它们是不同的源。
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**